pkg/vsphere/toolbox/hgfs/server.go

```diff
@@ -106,8 +109,75 @@ func (s *Server) Dispatch(packet []byte) ([]byte, error) {
 	return req.Reply(res, err)
 }
 
+// File interface abstracts standard i/o methods to support transfer
+// of regular files and archives of directories.
+type File interface {
+	io.Reader
+	io.WriteCloser
+
+	Name() string
+}
+
+// OpenFile selects the File implementation based on file type and mode.
+func (s *Server) OpenFile(name string, mode int32) (File, error) {
+	var err error
+	var file File
+
+	info, err := s.Stat(name)
+	if err == nil {
+		if _, ok := info.(*archive); ok {
+			switch mode {
+			case OpenModeReadOnly:
+				return newArchiveFromGuest(name)
+			case OpenModeWriteOnly:
+				return newArchiveToGuest(name)
+			}
+		}
+	}
+
+	switch mode {
+	case OpenModeReadOnly:
+		file, err = os.Open(name)
+	case OpenModeWriteOnly:
+		flag := os.O_WRONLY | os.O_CREATE | os.O_TRUNC
+		file, err = os.OpenFile(name, flag, 0600)
+	default:
```

Comment:
presume we hit this if `!s.Archive && info.IsDir()`? Should we be returning the VIX_E_NOT_A_FILE error you mention in the comment below?
Ah - I see it's translated in vix_command.go